driver-core/src/main/java/com/datastax/driver/core/querybuilder/Delete.java

```diff
@@ -39,6 +39,15 @@
         this.where = new Where(this);
         this.usings = new Options(this);
         this.conditions = new Conditions(this);
+
+        // This is for JAVA-1089, if the query deletes an element in a list, the statement should be non-idempotent.
+        for (Selector sel : columns) {
+            if (sel instanceof CollectionElementSelector) {
+                if (((CollectionElementSelector) sel).key instanceof Integer) {
```

Comment:
I believe this is wrong: it would also return true for maps with integer keys. We will need to create separate subclasses or find another way to distinguish list element deletions from other cases.